servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublisherFlatMapMerge.java

```diff
@@ -357,13 +332,8 @@ private void enqueueAndDrain(Object item) {
         }
 
         private void enqueueFailed(Object item) {
-            QueueFullException exception = new QueueFullException("signals");
-            if (item instanceof TerminalNotification) {
-                LOGGER.error("Queue should be unbounded, but an offer failed!", exception);
-                throw exception;
-            } else {
-                onError0(exception, false);
-            }
+            LOGGER.error("Queue should be unbounded, but an offer failed for item {}!", item);
+            throw new QueueFullException("pending");
```

Comment:
We discussed [here](https://github.com/apple/servicetalk/pull/1326#discussion_r560579593) that we can't throw inside terminal methods. But this one may potentially throw (if the queue changes). Consider using try-catch where this method is invoked.

Reply:
If we fail to enqueue a terminal event the async control flow won't be completed and the user won't be notified. This is a relatively extreme failure condition where I think it is justified to fail loudly as we are already violating the spec in more severe ways (e.g. not delivering all signals) and the behavior at this point is undefined (e.g. likely a hang). Note we were previously throwing in this scenario as well.

Reply:
I'll add a comment